### DISCUSSION TITLE
files with same names in a project require you to supply path

### DISCUSSION THREAD

**Discussion by jimkohl** [LefterisXris/CodeTour#88] [Ideas]
when adding a step by right clicking in a python repo, I was refeencing a __init__.py file.
Since its not unique within the project - I had to specif a specific path.  

However I'm already in the project and the path should be known by the IDE plugin - 
therefore should not have to ask for this path.

**Comment by LefterisXris** [comment]
Let me try to re-phrase what you said, to make sure we are on the same page:
- You create a Step on o file, which name may exist in the project on another package (init.py in your case)
- You suggest the CodeTour plugin could detect that, and automatically attach the path to the created step

If this is the case, it seems very reasonable to me, and yes I think it should work this way

**Reply by jimkohl** [reply]
yep thats what I meant

**Comment by sefi-monday** [comment]
Hi,

I second this - using the webstorm plugin I also expect it to know which `index.ts` file I am on when adding a new step.
Also - absolute paths entered manually does work (for example on mac `/Users/myself/Development/project/src/feature/index.ts` but that is not good enough for a team - each team member will have a different absolute path... so maybe also support relative paths - and define relative to where? the *.tour file? the project root?

**Reply by LefterisXris** [reply]
Hi @sefi-monday! Valid point.. Thanks for the feedback..

- The plugin could definitely get the active file instead of searching for duplicates. It's a good idea for improvement - I can work on it (or feel free to open PR maybe? :) )
- Relative paths can work at any level. Feel free to try it

**Reply by sefininio** [reply]
Hi @LefterisXris
I did try relative, but it didn't work for me...
Should it be relative to what? The project root? The tour file? I tried both without success.
Do you have it documented or can provide hints? Google didn't find anything unfortunately...

**Reply by LefterisXris** [reply]
Hi @sefininio. 

Relative paths are supported and they can work at any level ✅

The logic for finding the target file that is defined in the `Step.file` is to try matching the configured value using `endsWith` against the full path of existing virtual files (while looping in project files).

e.g. you can set in your tour Step:
```json
...
   {
      "title": "Code README",
      "description": "Simple Navigation to Code README.md",
      "file": "java/org/uom/lefterisxris/codetour/tours/README.md",
      "line": 1
    }
...
```

Here is the demonstration I did:
- I created a new branch `relative_branch_demo` with a full example for your convenience
- Created a new tour named `relativePathDemo.tour`
- Added a couple of duplicate files in my sources code (I used the README.md file)
- I added multiple steps in the new tour, where
  - One is prompting user for selection (as multiple matches are found)
  - One uses relative path from the parent dir
  - One used relative path from the source code
  - and similarly a couple of more...

Check out my commit for more details (make sure you check the full **tour**):  https://github.com/LefterisXris/CodeTour/commit/22c0eceb288715e735cbc6c4ccdbc9bb5d331485

**Reply by sefininio** [reply]
Awesome! That works and was a great help!
Thanks for the info 😀
